pkgs/mime/tool/mime_types.dart

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:mime/mime.dart';
+import 'package:mime/src/default_extension_map.dart';
+
+void main(List<String> args) {
```

Comment:
Can remove `List<String> args` when it isn't used.

(Could also use it, and allow a first argument to point to the file to update.
That would allow us to create a test for whether this code should be run by
making a copy of the `doc/mimetypes.md` file, calling this `main` with the path of the copy, and then checking if the files differ. Which I think we should do, so when you run tests, you're told if you need to update the doc.
If we do take the file on input, we should do better validation of it, like require it to exist, end in `.md`, and contain two markers, to avoid overwriting a wrong file.)

Reply:
For other similar systems, I've either had this script take an option `--verify` flag, which will fail if it would have written out differences, and run that on a CI, or, have the CI run the generate script, and then fail if `git diff` shows any differences.

Likely the 2nd would work here? It's pretty simple and will ensure that the file is always up-to-date.